scraper/src/libretexts2zim/client.py

```diff
@@ -0,0 +1,135 @@
+import datetime
+import re
+from collections.abc import Callable
+
+import requests
+from bs4 import BeautifulSoup, NavigableString
+from pydantic import BaseModel
+
+from libretexts2zim.constants import logger
+
+HTTP_TIMEOUT_SECONDS = 15
+
+
+class LibreTextsParsingError(Exception):
+    pass
+
+
+class LibreTextsShelve(BaseModel):
+    title: str
+    content_url: str
+    image_url: str
+
+
+class LibreTextsHome(BaseModel):
+    welcome_text_paragraphs: list[str]
+    shelves: list[LibreTextsShelve]
+    welcome_image_url: str
+
+
+class LibreTextsMetadata(BaseModel):
+    """Metadata about a course."""
+
+    # Human readable name for the course.
+    name: str
+
+    # URL prefix for the course, e.g. for Geosciences which is at
+    # https://geo.libretexts.org/, the slug is `geo`
+    slug: str
+
+    def placeholders(
+        self, clock: Callable[[], datetime.date] = datetime.date.today
+    ) -> dict[str, str]:
+        """Gets placeholders for filenames.
+
+        Arguments:
+          clock: Override the default clock to use for producing the "period".
+        """
+
+        return {
+            "name": self.name,
+            "slug": self.slug,
+            "clean_slug": re.sub(r"[^.a-zA-Z0-9]", "-", self.slug),
+            "period": clock().strftime("%Y-%m"),
+        }
+
+
+class LibreTextsClient:
+    """Utility functions to read data from libretexts."""
+
+    def __init__(self, library_slug: str) -> None:
+        """Initializes LibreTextsClient.
+
+        Paremters:
+            library_url: Scheme, hostname, and port for the Libretext library
+                e.g. `https://geo.libretexts.org/`.
```

Comment:
Where did you find reference of `course_prefix`?

I chose to not use a reference to `course` but `library` because looking at libretexts.org, it looks like they use the `library` word to reference a subsite like https://geo.libretexts.org/ (even if it is a bit "blurry"), and `course` word is anyway wrong since there are many courses on a single `library` like https://geo.libretexts.org/.

Reply:
```
    # URL prefix for the course, e.g. for Geosciences which is at
    # https://geo.libretexts.org/, the slug is `geo`
```
Library seems better indeed ; as long as we use the same wording everywhere 😉

Reply:
Fixed !